shm/Game.cpp

```diff
@@ -0,0 +1,75 @@
+#include "Game.hpp"
+
+#include <iostream>
+
+constexpr size_t distancePerDay{2};
+
+Game::Game(size_t money, size_t days, size_t final_goal)
+    : money_(money), days_(days), final_goal_(final_goal) {}
+
+bool Game::checkWinCondition() const {
+    return (player_->getMoney() >= final_goal_);
+}
+
+bool Game::checkLoseCondition() const {
+    return (player_->getMoney() == 0);
+}
```

Comment:
We should assure that this condition is checked after we left the Store, not after each transaction in Store. It is to avoid the situation when after buying some Cargo and before selling another one, player would have zero money.